impeller/renderer/backend/vulkan/pipeline_library_vk.cc

```diff
@@ -418,6 +437,26 @@ std::unique_ptr<PipelineVK> PipelineLibraryVK::CreatePipeline(
     auto vk_desc_layout = ToVKDescriptorSetLayoutBinding(layout);
     desc_bindings.push_back(vk_desc_layout);
   }
+  // Add subpass dependency binding.
+  if (desc.GetHasSubpassDependency()) {
+    vk::DescriptorSetLayoutBinding binding;
+    // _record scratch, freeze frame_
+    //
+    // I bet you're wondering where I got this number? Well its one less than
+    // the smallest fragment binding listed in the generated header for the
+    // framebuffer fetch shader. In order to look this up for real, we need to
+    // add support for this data to impellerc, but more importantly, we need to
+    // resolve trying to share a generated header across all platforms. Only the
+    // header generated from a shader compiled with Vulkan semantics will have
+    // this binding, but today its unspecified if we use the GLES or Metal
+    // version based on host platform. To that end, it would be safer to leave
+    // this as a hardcoded value until such a time as the bindings are fixed.
```

Comment:
I'm running into a similar issue looking at the fragment program API.

For this specific case, I think we could add another value to the generated header for this and only use it when we're in the Vulkan backend, right? IOW, if generated with vulkan semantics, the number is "real", otherwise it's just a placeholder.

However, I think to do runtime stage multi-plat right we'll want to have multiple compiler backends producing output. That can probably be limited just to runtime stage stuff, but maybe it's better to extend it to all the platforms.

Reply:
The problem is that subpass input is a vulkan only concept. if the header is GLSL or Metal generated, you won't have it. Which generated header do we use? It doesn't seem well defined. FWIW I will make the compiler output this value and see if it happens to work, but it still feels like a landmine.

We need different headers per platform/driver

Reply:
This is the same problem that has prevented the usage of half precision on Vulkan.